homeassistant/components/ring/binary_sensor.py

```diff
@@ -48,8 +51,8 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Ring binary sensors from a config entry."""
-    ring = hass.data[DOMAIN][config_entry.entry_id][RING_API]
-    devices = hass.data[DOMAIN][config_entry.entry_id][RING_DEVICES]
+    ring: Ring = hass.data[DOMAIN][config_entry.entry_id][RING_API]
+    devices: RingDevices = hass.data[DOMAIN][config_entry.entry_id][RING_DEVICES]
     notifications_coordinator: RingNotificationsCoordinator = hass.data[DOMAIN][
         config_entry.entry_id
     ][RING_NOTIFICATIONS_COORDINATOR]
```

Comment:
That's done, I went for a `dataclass` in the end because the `TypedDict` didn't seem to like the `CONSTANTS`.  In the end that allowed me to drop those constant values altogether which was nice.

Reply:
Yeah, to use constants as TypedDict keys, those need to be marked as `Final`. The `dataclass` works as well.